lib/streamlit/hashing.py

```diff
@@ -230,11 +230,21 @@ def is_simple(obj):
     return NoResult
 
 
+def _get_fqn_of_type(the_type):
+    """Get module.type_name for a given type."""
+    module = the_type.__module__
+    name = the_type.__name__
+    return "%s.%s" % (module, name)
+
+
 class _CodeHasher:
     """A hasher that can hash code objects including dependencies."""
 
     def __init__(self, hash_funcs=None):
-        self._hash_funcs = hash_funcs or {}
+        if hash_funcs:
```

Comment:
include note saying why we have to do this: user modules get reloaded bla bla.. see note from Thiago in eng doc

Reply:
Nit: I think this would be clearer

```
        # Can't use types as the keys in the internal _hash_funcs because
        # we always remove user-written modules from memory when rerunning a
        # script in order to reload it and grab the latest code changes.
        # (See LocalSourcesWatcher.py:on_file_changed) This causes
        # the type object to refer to different underlying class instances each run,
        # so type-based comparisons fail. To solve this, we use the types converted
        # to fully-qualified strings as keys in our internal dict.
```